wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/provisioncredential/EvidenceRequest.kt

```diff
@@ -102,6 +109,77 @@ fun EvidenceRequestMessageView(
     }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun EvidenceRequestNotificationPermissionView(
+    evidenceRequest: EvidenceRequestNotificationPermission,
+    provisioningViewModel: ProvisioningViewModel,
+    issuingAuthorityRepository: IssuingAuthorityRepository,
+    credentialStore: CredentialStore
+) {
+
+    // Only need to request POST_NOTIFICATIONS permission if on Android 13 (Tiramisu) or later.
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+        provisioningViewModel.provideEvidence(
```

Comment:
I am surprised that this works - you basically do the action in declarative @Composable context? I think you are not supposed to do that, it should either be resolved in the model itself, or something from https://developer.android.com/jetpack/compose/side-effects used here, perhaps just SideEffect { ... }

Reply:
Oh, not that part, provisioningViewModel.provideEvidence is imperative code executed in declarative context/environment

Reply:
Fixed using `SideEffect` as discussed on the chat, along with a `TODO` item that this needs to be revisited.